lxd/device/cdi/configure.go

```diff
@@ -0,0 +1,292 @@
+package cdi
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"tags.cncf.io/container-device-interface/specs-go"
+
+	"github.com/canonical/lxd/lxd/instance"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+// specDevToNativeDev builds a list of unix-char devices to be created from a CDI spec.
+func specDevToNativeDev(configDevices *ConfigDevices, d *specs.DeviceNode) error {
+	if d == nil {
+		return fmt.Errorf("CDI device node is nil")
+	}
+
+	if d.HostPath == "" && d.Path == "" {
+		return fmt.Errorf("Both hostPath and path are empty in the CDI device node: %v", *d)
+	}
+
+	var devPath string
```

Comment:
can you add some more comments here as its not clear why this is logic is needed.

Reply:
fixed